### PR TITLE
Fix Int/Float typecast issue.

### LINKED ISSUE #2259: Type coercion truncates Float to Int instead of raising an error
Hi guys; I'm pretty new to graphQL; I noticed it automatically coerces my type-float mutation arguments into type-int by truncation without raising any error. That seems to contradict what it says in the specs:

From https://graphql.github.io/graphql-spec/June2018/#sec-Int
> Input Coercion
> When expected as an input type, only integer input values are accepted. All other input values, including strings with numeric content, must raise a query error indicating an incorrect type.

in https://github.com/rmosolgo/graphql-ruby/blob/master/lib/graphql/types/int.rb#L13
>      def self.coerce_input(value, _ctx)
>        value.is_a?(Numeric) ? value.to_i : nil
>      end

should it not be
> value.is_a?(Integer) ? value.to_i : nil

instead?

### PR DESCRIPTION
**Pull request by cherbst-2112**
Closes #2259

Raise an error if an Integer argument receives a Float.

### CODE REVIEW

**Comment by cherbst-2112** [comment]
n/m, actually looks like that may be by design:

```

    describe "scalar inputs" do
      let(:query_string) {%|
        {
          cheese(id: 1.4) {
            flavor
          }
        }
      |}

      it "converts them to the correct type" do
        cheese_name = result["data"]["cheese"]["flavor"]
        assert_equal("Brie", cheese_name)
      end
    end
  end
```

**Comment by rmosolgo** [comment]
Personally, I think youre right about that. Here's the spec: 

> _Input Coercion_
>
> When expected as an input type, only integer input values are accepted. All other input values, including strings with numeric content, must raise a query error indicating an incorrect type.

https://graphql.github.io/graphql-spec/June2018/#sec-Int

Doesn't it sound like your change is correct?

**Comment by cherbst-2112** [comment]
Huh, OK, noticed the affected tests were mostly typecasting IDs as floats which *really* doesn't make sense anyway, they should be ID/String types.

I updated the affected tests, let me know what you think!

File: lib/graphql/types/int.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@ class Int < GraphQL::Schema::Scalar
       MAX = (2**31) - 1
 
       def self.coerce_input(value, _ctx)
-        value.is_a?(Numeric) ? value.to_i : nil
+        value.is_a?(Integer) ? value.to_i : nil
```
Comment (rmosolgo) [inline lib/graphql/types/int.rb]: Do we still need to call `.to_i` in the case that it's an Integer? Or could it be: 

```suggestion
        value.is_a?(Integer) ? value : nil
```

**Comment by cherbst-2112** [comment]
Thanks! I've made that change, but I'm getting some unrelated network errors in travis

```
[1/4] Resolving packages...
error An unexpected error occurred: "https://registry.yarnpkg.com/apollo-link: ETIMEDOUT".
info If you think this is a bug, please open a bug report with the information provided in "/home/travis/build/rmosolgo/graphql-ruby/javascript_client/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
yarn run v1.15.2
$ jest
/bin/sh: 1: jest: not found
error Command failed with exit code 127.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

I'll try again later or tomorrow

**Comment by rmosolgo** [comment]
I've been seeing that too -- It seems like there's some network-level (caching?) issue. I re-ran the failed runs and they passed. 

thanks for your work on this!

**Pull request merged**